File: src/CtrDxEditor.Browser/wwwroot/webcrypto.js
// SHA-256 over a byte array using the browser's native SubtleCrypto, which runs
// off the main thread. Returns lowercase hex to match the desktop hashing path.
export async function sha256Hex(bytes) {
    const digest = await crypto.subtle.digest("SHA-256", bytes);
    return [...new Uint8Array(digest)]
        .map((b) => b.toString(16).padStart(2, "0"))
        .join("");
}

File: src/CtrDxEditor.Browser/Content/WebCrypto.cs
using System.Runtime.InteropServices.JavaScript;
using System.Threading.Tasks;

namespace CtrDxEditor.Browser.Content
{
    /// <summary>Thin managed wrapper over the webcrypto.js SubtleCrypto module.</summary>
    internal static partial class WebCrypto
    {
        /// <summary>Imports the webcrypto.js module. Must be awaited once before any other call.</summary>
        public static Task ImportAsync()
        {
            return JSHost.ImportAsync("webcrypto", "../webcrypto.js");
        }

        /// <summary>Computes the lowercase-hex SHA-256 of the given bytes off the main thread.</summary>
        [JSImport("sha256Hex", "webcrypto")]
        public static partial Task<string> Sha256HexAsync(byte[] bytes);
    }
}
